extension/js/common/core/crypto/key.ts

```diff
@@ -184,7 +184,9 @@ export class KeyUtil {
     if (!exp) {
       return false;
     }
-    if (exp instanceof Date) {
+    // exp instanceof Date does not work if the date objects, are from another realm (e.g. iframe)
+    // see https://github.com/FlowCrypt/flowcrypt-browser/pull/2816
+    if (typeof exp.getTime === 'function') {
```

Comment:
Here you will need to disable a linter check for tests to pass. Also, the check here is different from the check in openpgp-key, is that by design?

Reply:
When I do very verbose checks like this, it is typically because I used to get some other value in the past and I didn't know what it was (regardless of the types show - after all, weird stuff can happen in real life that the type system may not know about).

But I'll merge this, maybe the reason I added that is no longer relevant.